src/Credit/Client.php

```diff
@@ -70,14 +61,55 @@ public function getConfig(): Platon\ConfigInterface
         return $this->config;
     }
 
+    protected function generateParams(Credit\TransferInterface $creditToCard): array
+    {
+        $this->validateCardToken($creditToCard->getCardToken());
+
+        $params = [
+            'client_key' => $this->config->getKey(),
+            'order_currency' => $creditToCard->getCurrency(),
+            'order_description' => $creditToCard->getDescription(),
+            'order_id' => $creditToCard->getId(),
+            'order_amount' => number_format($creditToCard->getAmount(), 2, '.', ''),
+        ];
+
+        $token = $creditToCard->getCardToken();
+
+        if ($this->isCardNumber($token)) {
+            $params['action'] = CreditToCardInterface::ACTION_CARD;
+            $params['card_number'] = $token;
+            $this->appendCardHash($params);
+
+            if ($creditToCard instanceof HasExpireDate) {
+                $params['card_exp_month'] = $creditToCard->getExpireMonth();
+                $params['card_exp_year'] = $creditToCard->getExpireYear();
+            }
+        } else {
+            $params['action'] = CreditToCardInterface::ACTION_TOKEN;
+            $params['card_token'] = $token;
+            $this->appendTokenHash($params);
+        }
+
+        return $params;
+    }
+
+    protected function isCardNumber(string $token): bool
+    {
+        return is_numeric($token) && $this->luhn->isValid(LuhnAlgorithm\Number::fromString($token));
+    }
+
     protected function validateCardToken(string $cardToken): void
     {
+        if ($this->isCardNumber($cardToken)) {
+            return;
```

Comment:
\InvalidArgumentException have to be thrown.
Calling this method with card number as argument is invalid case